### ISSUE TITLE
content detection deprecated warning

### ISSUE THREAD

**Issue by zh217** [mpenet/spandex#19]
getting lots of these:

```
WARN [org.elasticsearch.client.RestClient:65] - nil request [PUT http://127.0.0.1:9200/_bulk] returned 1 warnings: [299 Elasticsearch-5.4.0-780f8c4 "Content type detection for rest requests is deprecated. Specify the content type using the [Content-Type] header." "Sat, 03 Jun 2017 17:17:37 GMT"]
```

**Comment by mpenet** [comment]
Can you show me the call that triggers these?

**Comment by zh217** [comment]
Sure. Any request with a non-empty body will do

```clj
(s/request
  client
  {:url     (s-utils/url [:_search])
   :method  :get
   :body    {:from  0
             :size  1
             :query {:match_all {}}}})

#=> 299 Elasticsearch-5.4.0-780f8c4 "Content type detection for rest requests is deprecated. Specify the content type using the [Content-Type] header." "Sun, 04 Jun 2017 10:18:59 GMT"
```

Using `s/bulk-chan` will also generate the same warning.

If I set the headers manually there's no warning:

```clj
(s/request
  client
  {:url     (s-utils/url [:_search])
   :method  :get
   :headers {:Content-Type "application/json; charset=UTF8"}
   :body    {:from  0
             :size  1
             :query {:match_all {}}}})
```

however there's no way to set headers using `s/bulk-chan`.

**Comment by mpenet** [comment]
You can pass a request map to bulk-chan like request & co. 

Odd thing is that I set the ct in that function, you shouldnt get any warning. I ll check this on monday when I am back.

**Comment by mpenet** [comment]
Maybe ES is case sensitive (I set it in  lowercase)

**Comment by zh217** [comment]
Thanks for the reply.

It turns out that ES is not case sensitive, but for bulk requests it expected a content-type of `application/x-ndjson`, whereas in `s/bulk-chan` the given type was `text/plain`. That's why ES complained.

Setting the correct content-type for `s/bulk-chan` suppresses the warnings.

**Comment by mpenet** [comment]
Fixed in 0.4.0 (it's on clojars). 

Thanks
